### PR TITLE
Container tests and refactors

### PR DESCRIPTION
**Pull request by mpeck**
Adds some tests to the container commands and refactors things a bit.

### CODE REVIEW

File: spec/cog_cmd/ecs/container/update_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    }
+    END
+    expect(client).to receive(:put_object).with(bucket: 'fakebucket', key: key, body: body_str.chomp)
+
+    run_command(args: ['mydef'], options: { image: 'myupdatedimage' })
+
+    expect(command).to respond_with({ 'name' => 'mydef',
+                                      'image' => 'myupdatedimage',
+                                      'memory' => 256 })
+  end
```
Comment (vanstee) [inline spec/cog_cmd/ecs/container/update_spec.rb]: @mpeck Some of these tests do a lot of stubbing/mocking which is usually a negative sign. We don't have to fix that right away, but it's a sign that we should probably introduce more objects, like the `Client` stuff we talked about, so we have less dependencies to stub.

File: spec/cog_cmd/ecs/container/update_spec.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
+require 'spec_helper'
+
+describe 'the container-update command' do
+  let(:command_name) { 'container-update' }
+
+  let(:client) do
+    client = Object.new
+    allow(Aws::S3::Client).to receive(:new).and_return(client)
```
Comment (vanstee) [inline spec/cog_cmd/ecs/container/update_spec.rb]: This is probably better suited for a `before` block. The result would look something like this.

```
let(:command_name) { 'container-update' }
let(:client) { Object.new }

before do
  allow(Aws::S3::Client).to receive(:new).and_return(client)
end
```

File: spec/cog_cmd/ecs/container/update_spec.rb (hunk 1 above)
Comment (mpeck) [inline spec/cog_cmd/ecs/container/update_spec.rb]: Yeah, I agree.

File: spec/cog_cmd/ecs/container/update_spec.rb (hunk 2 above)
Comment (mpeck) [inline spec/cog_cmd/ecs/container/update_spec.rb]: Oh yeah, that is nicer.

File: spec/cog_cmd/ecs/container/list_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  let(:client) do
+    client = Object.new
+    allow(Aws::S3::Client).to receive(:new).and_return(client)
+    client
+  end
+
+  let(:object_list) do
+    obj = Aws::S3::Types::Object.new(key: 'container-definitions/mydef.json',
+                                     last_modified: '2016-09-06 23:39:15 UTC')
+    Aws::S3::Types::ListObjectsV2Output.new(contents: Array.new.push(obj))
```
Comment (vanstee) [inline spec/cog_cmd/ecs/container/list_spec.rb]: I think you can use an array literal here: `Aws::S3::Types::ListObjectsV2Output.new(contents: [obj])`

File: spec/cog_cmd/ecs/container/define_spec.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    container_def = double(body: double(read: '{"name":"mydef","memory":256,"image":"myimage"}'))
+
+    expect(client).to receive(:put_object)
+    expect(client).to receive(:get_object).and_return(container_def)
+
+    run_command(args: ['mydef', 'myimage'], options: { memory: 256 })
+
+    expect(command).to respond_with({ 'name' => 'mydef',
+                                      'image' => 'myimage',
+                                      'memory' => 256 })
```
Comment (vanstee) [inline spec/cog_cmd/ecs/container/define_spec.rb]: Is the `command` and `respond_with` stuff injected by cog-rb? Feels a little magical ✨

Comment (mpeck) [inline spec/cog_cmd/ecs/container/define_spec.rb]: It is yes. We can discuss changing the cog-rb test stuff at some point. Honestly it seemed fine for now, so I just let it be.

Comment (vanstee) [inline spec/cog_cmd/ecs/container/define_spec.rb]: WFM :+1:

**Comment by vanstee** [comment]
Lookin' good. :+1:

**Pull request merged**